src/lib.rs

```diff
@@ -1,19 +1,107 @@
+/*!
+  # Selecting the right chip
+
+    This crate requires you to specify your target chip as a feature.
+
+    Please select one of the following
+
+    (Note: `x` denotes any character in [a-z])
+    *   stm32f301xb
+    *   stm32f301xc
+    *   stm32f301xd
+    *   stm32f301xe
+    *   stm32f318
+    *   stm32f302
+    *   stm32f303xb
+    *   stm32f303xc
+    *   stm32f303xd
+    *   stm32f303xe
+    *   stm32f303x6
+    *   stm32f303x8
+    *   stm32f373
+    *   stm32f378
+    *   stm32f334
+    *   stm32f328
+    *   stm32f358
+    *   stm32f398
+
+    Example: The STM32F3Discovery board has a STM32F303VCT6 chip.
+    So you want to expand your call to `cargo` with `--features stm32f303xc`.
+
+    For more information, see the [README](https://github.com/stm32-rs/stm32f3xx-hal/blob/master/README.md#selecting-the-right-chip)
+ */
 #![no_std]
 #![allow(non_camel_case_types)]
 
-#[cfg(not(feature = "device-selected"))]
+#[cfg(all(not(feature = "device-selected"), not(feature = "needs-subvariant")))]
```

Comment:
Revert this. Currently, there is no device, which simultaneously has `device-selected` and `needs-subvariant` and it is also not planned in the future :)

Reply:
if you use `--features stm32f318` you only get `device-selected`.
if you use `--features stm32f303` you only get `needs-subvariant`.
if you use `--features stm32f303xc` you will get `device-selected` and `stm32f303`. by getting `stm32f303` you'll get `needs-subvariant` indirectly.
Also having neither `device-selected` nor `needs-subvariant` means you simply didnt specify any chip.
So all combinations are possible.